pkg/cloudprovider/vm.go

```diff
@@ -105,6 +105,10 @@ func (i *vmIdentityInfo) RemoveUserIdentity(id string) error {
 	if i.info.Type == compute.ResourceIdentityTypeNone || i.info.Type == compute.ResourceIdentityTypeSystemAssigned {
 		i.info.IdentityIds = nil
 	}
+	// if the identityids is nil and identity type is not set, then set it to ResourceIdentityTypeNone
+	if i.info.IdentityIds == nil && i.info.Type == "" {
```

Comment:
Thinking more here, shouldn't we just skip the call to cloud overall in this scenario ?